client/consensus/common/src/tests.rs

```diff
@@ -187,6 +214,68 @@ fn follow_new_best_works() {
 	});
 }
 
+#[test]
+fn follow_new_best_with_dummy_recovery_works() {
```

Comment:
Test to excercise explicit best block recovery.

Even though currently we are using as a recovery mechanism the `PoVRecovery` subsystem, the motivation of this test is to also show how the recovery can be generic.